ouroboros-consensus-test/test-consensus/Test/Consensus/MiniProtocol/BlockFetch/Client.hs

```diff
@@ -193,19 +231,52 @@ runBlockFetchTest BlockFetchClientTestSetup{..} = withRegistry \registry -> do
                       case Chain.applyChainUpdates updates chain of
                         Just chain' -> chain'
                         Nothing     -> error "Chain update failed"
-                -- Block until our "ChainSync" thread registered itself to the
-                -- FetchClientRegistry, see 'forkChainSync' below.
-                _ <- blockUntilJust $ Map.lookup peerId <$> readTVar varChains
+                    updateCPS cps =
+                      case CPS.applyChainUpdates (fmap TestHeader <$> updates) cps of
+                        Just cps' -> cps'
+                        Nothing   -> error "Chain update failed"
                 modifyTVar varChains $ Map.adjust updateChain peerId
+                modifyTVar (producerStateVars Map.! peerId) updateCPS
 
         forkChainSync peerId =
-          forkLinkedThread registry ("BracketSync" <> condense peerId) $
+            forkThread registry ("BracketSync" <> condense peerId) $
```

Comment:
Actually, I have the same question about the BlockFetch clients too, I suppose.

Reply:
Sounds also like a possible approach, I am not necessarily opposed. I think the main thing here is that we then are no longer able to independently see whether ChainSync *or* BlockFetch threw an exception (i.e. we only have `Either ChainSyncEx BlockFetchEx` instead of `These ChainSyncEx BlockFetchEx`). Right now, we e.g. use this information for labelling, see `tabulateFailureMode`, which results e.g. in
```
      Expected failure due to (3388 in total):
      64.14% ChainSync
      34.65% BlockFetch,ChainSync
       1.21% BlockFetch
```